doc/Developer-checklist.md

```diff
@@ -87,3 +87,67 @@ If you can, aim for long pipelines of processing operations.
 If you can, put large resizes right at the start (see `thumbnail` above),
 then area filters (sharpen, for example), and finally any point operations.
 
+## Only enable the load libraries you need
+
+libvips after version 8.13 has a system for enabling and disabling image load
+libraries at runtime, see:
+
+https://www.libvips.org/2022/05/28/What's-new-in-8.13.html
+
+You can usually improve security and avoid memory spikes by only enabling
+the image formats you really need.  If you are handling untrusted data,
+I would set the `VIPS_BLOCK_UNTRUSTED` env var and only use the loaders we
+have tested for security.
+
+Older versions of libvips need compile-time configuration.
+
+## Sanity-check images before processing
+
+libvips image open is always fast and safe, as long as you have disabled
+load via imagemagick. This means you can open an image and sanity-check it
+before further processing.
+
+There are two main checks that are very worthwhile:
+
+1. Sanity check image dimensions to protect you from decompression
+   bombs like those described at
+   https://www.bamsoftware.com/hacks/deflate.html
+
+2. Check for interlaced (also called progressive) images.
+
+   These are the ones that appear at a low res first, then slowly fill in
+   detail as they are downloaded.
+
+   The downside is that you don't get the final pixels until the whole image
+   is in memory, which prevents any streaming processing and hugely increases
+   memory use. For example:
+
+```
+$ /usr/bin/time -f %M:%e vipsthumbnail big-progressive.jpg
+3732224:4.23
+$ vips copy big-progressive.jpg x.jpg
+$ /usr/bin/time -f %M:%e vipsthumbnail x.jpg
+72448:0.26
+$
```

Comment:
This stray cmd line can be removed
```suggestion
```
(perhaps this entire cmd block should also be indented with 3 spaces?)